core/pkg/server/sender.go

```diff
@@ -214,7 +215,13 @@ func (s *Sender) Do(allWork <-chan runwork.Work) {
 	defer s.logger.Reraise()
 	s.logger.Info("sender: started", "stream_id", s.settings.RunId)
 
+	hangDetectionInChan := make(chan runwork.Work, 32)
+	hangDetectionOutChan := make(chan struct{}, 32)
+	go s.warnOnLongOperations(hangDetectionInChan, hangDetectionOutChan)
+
 	for work := range allWork {
```

Comment:
i don't know if it is a good idea, but maybe having it built in into our work channels could be cool